script.js

```diff
@@ -40,15 +44,28 @@ function updateButtonStatus() {
   actionButtonBack.style.visibility = 'visible'
   actionButtonNext.textContent = 'Next step'
   actionButtonNext.classList.remove('action-buttons__next--confirm')
-  if (globalActiveSectionNumber === 0) {
+  if (globalSectionIndex === 0) {
     actionButtonBack.style.visibility = 'hidden'
     actionButtonNext.classList.remove('action-buttons__next--confirm')
-  } else if (globalActiveSectionNumber === 3) {
+  } else if (globalSectionIndex === 3) {
     // change color and text
     actionButtonNext.classList.add('action-buttons__next--confirm')
     actionButtonNext.textContent = 'Confirm'
-  } else if (globalActiveSectionNumber === 4) {
+  } else if (globalSectionIndex === 4) {
     // hide all buttons componet
     actionButtonsDiv.classList.add('visually-hidden')
   }
 }
+
+//Update the steps status depends on section position
+function updateStepsStatus() {
+  stepsItems.forEach((item) => {
+    item.classList.remove('side-bar__number--active')
+  })
+
+  if (globalSectionIndex === 4) {
+    stepsItems[globalSectionIndex - 1].classList.add('side-bar__number--active')
+  } else {
+    stepsItems[globalSectionIndex].classList.toggle('side-bar__number--active')
```

Comment:
in both cases you are adding the class because you removed the class from all of the items in the foreach, for consistency  in this case you should be using either add or toggle, not both. Also I would have done something like
```
val index = globalSectionIndex === 4 ? globalSectionIndex - 1 : globalSectionIndex
stepsItems[globalSectionIndex].classList.add('side-bar__number--active')
```